File: LeetCode/69 - Sqrt(x)/solution.dart
class Solution {
  int mySqrt(int x) {
    var left = 0;
    var right = x;

    while (left <= right) {
      var mid = ((left + right) / 2).floor();

      if ((mid * mid) <= x) {
        left = mid + 1;
      } else {
        right = mid - 1;
      }
    }
    return left - 1;
  }
}

File: LeetCode/69 - Sqrt(x)/solution.py
class Solution:
    def my_sqrt(self, x: int) -> int:
        left, right = 0, x
        while left <= right:
            mid = (left + right) // 2
            if mid * mid <= x:
                left = mid + 1
            else:
                right = mid - 1
        return left - 1
